.github/workflows/install-rust-and-protoc/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Install Rust tool chain and protoc
```

Comment:
why moving it to a different action (e.g. making it a function) if it's only being called from a single action?

Reply:
I rather calling twice "${{ !contains(inputs.target, 'musl') }}" in the original action, once for protoc and once for rust, rather than maintaining another action file with changing inputs ('target')

Reply:
or you can move all of this change to a separate commit and refactor other actions to call this too (lint-rust, for example)

Reply:
also note that if you keep it you'll need to add this action path to to pull request triggers - so if we do changes in this file it will trigger the actions that depend on it. 
```
#npm-cd.yml for example
    pull_request:
      paths:
        - .github/workflows/npm-cd.yml
        - .github/workflows/build-node-wrapper/action.yml
        - .github/workflows/start-self-hosted-runner/action.yml
        - .github/workflows/install-rust-and-protoc/action.yml

Reply:
When any action start it is "preparing" the action. Its doing it even if its excluded for the specific target and its breaking the CI on musl CI process since its cant be done with musl on arm. So i had no choice but taking it out of the action so on musl it does't try to prepare it.  
I'll add it to the pull path, but for making this PR works this has to be merge first or with.